### PR TITLE
Adds deployment with capistrano and CI with travis

### PR DESCRIPTION
**Pull request by jgreben**
.travis.yml file generated by:
```
$ travis setup releases --force
Username: jgreben
Password for jgreben: *********
File to Upload: 
Deploy only from sul-dlss-labs/Vitro? |yes| 
Deploy from maint-rel-1.9 branch? |yes| 
Encrypt API key? |yes| 
```

There are no specific deploy tasks defined yet. Also note that the branch to trigger a build is:
```
on:
    repo: sul-dlss-labs/Vitro
    branch: maint-rel-1.9
```

Currently there are changes to this branch on the server so those will need to be merged first.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,21 @@
 language: java
 sudo: false
-
 env:
-  # Give Maven 1GB of memory to work with
-  - MAVEN_OPTS=-Xmx1024M
-
+- MAVEN_OPTS=-Xmx1024M
```
Comment (mjgiarlo) [inline .travis.yml]: It looks like this file may have had its formatting auto-cleaned up. I would have expected this line to be indented two spaces, which also applies to the lists below `directories`, `install`, and `script` in the following lines. Note that maybe this is still valid YAML syntax -- my :eyes: are not yet certified YAML validators.

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,21 @@
...
+cache:
+  directories:
+  - "$HOME/.m2/repository"
+before_install:
+- mvn compile
 install:
-  - "echo 'Skipping install stage, dependencies will be downloaded during build and test stages.'"
-
+- echo 'Skipping install stage, dependencies will be downloaded during build and test
+  stages.'
```
Comment (mjgiarlo) [inline .travis.yml]: This is one of those changes I mentioned above, where it seems like we may want the unchanged code where `- "echo...` is indented under `install:`, and the echo string is not broken across two lines.

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
 script:
-  - "mvn clean package -Dmaven.test.skip=false"
+- mvn clean package -Dmaven.test.skip=false
```
Comment (mjgiarlo) [inline .travis.yml]: Indent two spaces?

File: config/deploy.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
+set :application, 'Vitro'
+set :repo_url, 'https://github.com/sul-dlss-labs/Vitro.git'
+
+# Default branch is :master
+ask :branch, `git rev-parse --abbrev-ref HEAD`.chomp
+
+# Default deploy_to directory is /var/www/my_app_name
+set :deploy_to, "/opt/app/vitro/src/Vitro"
```
Comment (mjgiarlo) [inline config/deploy.rb:8]: @jgreben If I understand, this is where we make sure that our most recent code is reflected on our box. How would we trigger the `mvn install -s rialto-vitro-settings.xml` step on the box? Or is that not needed? (We'll also need to bounce tomcat.) Or do you see these steps as ripe for a follow-up PR?

**Review by mjgiarlo — COMMENTED** [review]
Left some comments about YAML formatting and some questions about the parts I don't understand. Thanks for getting this started!

File: .travis.yml (hunk 1 above)
Comment (jgreben) [reply .travis.yml]: Good 👁 Giarlo!

File: config/deploy.rb (hunk 4 above)
Comment (jgreben) [reply config/deploy.rb:8]: Yes, I was thinking a follow-up PR for whatever deploy tasks are needed. In the reference app I was looking at under `namespace :maven` we had the `:package` task. So a follow-up PR can specify a task or two to do the maven install and restart tomcat. I'm not sure about how the permissions might work with tomcat,  but I thought we could cross that bridge once we've got the basic deploy working.

Comment (mjgiarlo) [reply config/deploy.rb:8]: Sounds good, @jgreben

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**